src/server/models/okta/Error.ts

```diff
@@ -0,0 +1,87 @@
+export interface ErrorResponse {
+  errorCode: string;
+  errorSummary: string;
+  errorLink: string;
+  errorId: string;
+  errorCauses: Array<ErrorCause>;
+}
+
+export interface ErrorCause {
+  errorSummary: string;
+}
+
+export type ErrorCode = 'E0000001' | 'E0000007' | 'E0000016' | 'E0000038';
+
+export class OktaError extends Error {
+  name: string;
+  code?: ErrorCode;
+  constructor(message?: string, name = 'OktaError', code?: ErrorCode) {
+    super(message);
```

Comment:
Does `Object.setPrototypeOf(this, new.target.prototype);` need to be set after calling `super()` see - https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html#support-for-newtarget I'm not sure if its applicable

Reply:
I'm not sure, but the current IDAPI errors also extend Error without including this line, so if it were an issue it would already be affecting what is currently in prod

Reply:
👍